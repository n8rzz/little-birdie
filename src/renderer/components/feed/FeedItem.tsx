import React from 'react';
import { IFeed } from '../../../shared/domain/feed/models/i-feed';

interface IProps {
  item: IFeed;
}

export const FeedItem: React.FC<IProps> = (props: IProps) => {
  return (
    <li className="providerList-item">
      <a href={props.item.link}>{ props.item.name }</a>
    </li>
  );
};
