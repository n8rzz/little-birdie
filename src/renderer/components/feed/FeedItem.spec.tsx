import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react';
import { parsedRssMock } from '../../__mocks__/parsed-rss.mock';
import { FeedItem } from './FeedItem';

describe('FeedItem', () => {
  it('should match snapshot', () => {
    const tree = TestRenderer.create(<FeedItem item={parsedRssMock[0]} />);

    expect(tree).toMatchSnapshot();
  });

  describe('should render correct feed name', () => {
    const { getByText } = render(<FeedItem item={parsedRssMock[0]} />);

    expect(getByText(parsedRssMock[0].name)).not.toBeUndefined();
  });
});
