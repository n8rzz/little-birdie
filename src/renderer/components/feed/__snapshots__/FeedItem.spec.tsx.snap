// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedItem should match snapshot 1`] = `
<li
  className="providerList-item"
>
  <a
    href="https://www.harveststatus.com"
  >
    Harvest
  </a>
</li>
`;
